crates/transaction-emitter/src/nft_txn_emitter.rs

```diff
@@ -0,0 +1,189 @@
+};
+use diem_transaction_builder::experimental_stdlib;
+use diem_types::nft::tokens;
+use futures::future::try_join_all;
+use std::cmp::min;
+
+#[derive(Debug)]
+pub struct NFTTxnEmitter {
+    accounts: Vec<LocalAccount>,
+    txn_factory: TransactionFactory,
+    client: RestClient,
+    bars_account: LocalAccount,
+    rng: ::rand::rngs::StdRng,
+}
+
+impl NFTTxnEmitter {
+    pub fn new(
+        client: RestClient,
+        transaction_factory: TransactionFactory,
+        bars_account: LocalAccount,
+        rng: ::rand::rngs::StdRng,
+    ) -> Self {
+        Self {
+            accounts: vec![],
+            txn_factory: transaction_factory,
+            client,
+            bars_account,
+            rng,
+        }
+    }
+}
+
+#[async_trait]
+impl TxnEmitter for NFTTxnEmitter {
+    fn accounts(&mut self) -> &mut Vec<LocalAccount> {
+        &mut self.accounts
+    }
+
+    fn transaction_factory(&self) -> &TransactionFactory {
+        &self.txn_factory
+    }
+
+    fn client(&self) -> &RestClient {
+        &self.client
+    }
+
+    fn rng(&mut self) -> &mut ::rand::rngs::StdRng {
+        &mut self.rng
+    }
+
+    async fn mint_accounts(
+        &mut self,
+        _req: &EmitJobRequest,
+        total_requested_accounts: usize,
+    ) -> Result<()> {
+        if self.accounts.len() >= total_requested_accounts {
+            info!("Already have enough accounts exist, do not need to mint more");
+            return Ok(());
+        }
+
+        let num_accounts = total_requested_accounts - self.accounts.len(); // Only minting extra accounts
+        let bars_per_account = SEND_AMOUNT * MAX_TXNS;
+        let txn_factory = self.txn_factory.clone();
+
+        let mut i = 0;
+        while i < total_requested_accounts {
+            let batch_size = min(min(MAX_TXN_BATCH_SIZE, total_requested_accounts - i), 100);
+            let mut batch = gen_random_accounts(batch_size, self.rng());
+
+            // Create
+            let create_requests = batch
+                .as_slice()
+                .iter()
+                .map(|account| {
+                    gen_create_account_txn_request(&mut self.bars_account, account, &txn_factory)
+                })
+                .collect();
+            execute_and_wait_transactions(&self.client, &mut self.bars_account, create_requests)
+                .await?;
+
+            // register
+            try_join_all(batch.as_mut_slice().iter_mut().map(|account| {
+                let request = gen_register_account_txn_request(account, self.transaction_factory());
+                execute_and_wait_transactions(&self.client, account, vec![request])
+            }))
+            .await?;
+
+            // mint
+            let mint_requests = batch
+                .as_slice()
+                .iter()
+                .map(|account| {
+                    gen_mint_txn_request(
+                        &mut self.bars_account,
```

Comment:
It seems that right now for each account, we mint many copies of one single token, and transfer that only token that's in the user's gallery. This doesn't seem to be sufficient to measure the actual performance of transferring in reality? Right now the complexity of transferring a token can be `O(n)` where `n` is the number of distinct tokens in the transferrer's gallery. And this also depends on the index of the token to be transferred inside the gallery, since we currently use linear search to find the token. I'm wondering if it makes sense to add an option to specify `num_distinct_tokens_per_account` and maybe transfer a randomly selected token each time.